### ISSUE TITLE
vsf vs wfo 

### ISSUE THREAD

**Issue by Taydra-Low** [NOAA-GFDL/xwmt#44]
Some models (CESM2) only have virtual salt flux (vsf) and not wfo. Is the wfo converted into a vsf? Or is there functionality to work with a vsf with xwmt?

**Comment by hdrake** [comment]
That's a great question @Taydra-Low. Because the VSF is already formulated as a tracer flux that affects tracer content, it should be treated just like any other tracer tendency (as opposed to `wfo`, which is a mass flux exchange that only indirectly affects salt concentrations by diluting salt content).

This is easy to accommodate in my branch https://github.com/hdrake/xwmt but would require source code changes in the main or @jetesdal's branches.

**Comment by jetesdal** [comment]
Hi @Taydra-Low! Thanks for bringing this up. To answer your question, the main branch version does not consider virtual salt flux. The current main branch is somewhat legacy code from submission to [CMEC](https://cmec.llnl.gov/), and it is not actively developed. I agree with @hdrake to add the virtual salt flux in his branch. It is also our long-term goal to merge the development in Henri’s branch with the current main branch.

**Comment by Taydra-Low** [comment]
Hi!

Thanks for the speedy response. Yeah after looking through the code more deeply, it looks as though the mass flux (wfo) is converted to a salt flux (vsf) when computing the density flux. So, for preprocessing I will just have to convert the vsf to a wfo which then gets converted back into a vsf in the computation. I believe the conversion is vsf/-so. I have yet to look at Henri's branch, but I am curious now.

Cheers,
Taydra
________________________________

**Comment by hdrake** [comment]
> Yeah after looking through the code more deeply, it looks as though the mass flux (wfo) is converted to a salt flux (vsf) when computing the density flux. So, for preprocessing I will just have to convert the vsf to a wfo which then gets converted back into a vsf in the computation.

Yes, you could either address this in preprocessing as your suggest or you could change the `xwmt` source code to allow `vsf` to be treated directly (just like surface heat fluxes already are).

> I believe the conversion is vsf/-so. I have yet to look at Henri's branch, but I am curious now.
Happy to help you port your code over to my branch, if you want. Just open a separate Issue if you run into any problems.

**Comment by gmacgilchrist** [comment]
I'll just chime in here to say that it would be great to merge @hdrake's branch soon, because I think the API is generally much more flexible than what we currently have and because it's the relevant approach for the MS Henri just revised. It will be important not to break @Taydra-Low's code though (or anyone else who's making use of this), meaning that she will either need to continue using the old version, or Henri/others could help port it over to the new version.

**Comment by hdrake** [comment]
Yes, I would help @Taydra-Low port their code to the new version.

As discussed in [this comment](https://github.com/NOAA-GFDL/xwmt/pull/40#issuecomment-1700189638), the new version reproduces @jetesdal's version that Taydra is currently using (although my sign convention for the diagnosed water mass transformations has since changed).
